src/Features/Core/Portable/Navigation/INavigableItem.cs

```diff
@@ -45,5 +48,39 @@ internal interface INavigableItem
         bool IsStale { get; }
 
         ImmutableArray<INavigableItem> ChildItems { get; }
+
+        public record NavigableDocument(NavigableProject Project, string Name, string? FilePath, IReadOnlyList<string> Folders, DocumentId Id)
+        {
+            public required bool IsSourceGeneratedDocument { get; init; }
+            public required Workspace Workspace { get; init; }
```

Comment:
why have some methods be parameters, and some be required props?

Reply:
⏱️ Moving all to parameters